isis/src/base/objs/SpicePosition/SpicePosition.cpp

```diff
@@ -382,25 +373,31 @@ namespace Isis {
     p_fullCacheSize = isdPos["SpkTableOriginalSize"].get<double>();
     p_cacheTime = isdPos["EphemerisTimes"].get<std::vector<double>>();
 
-    for (auto it = isdPos["Positions"].begin(); it != isdPos["Positions"].end(); it++) {
-      std::vector<double> pos = {it->at(0).get<double>(), it->at(1).get<double>(), it->at(2).get<double>()};
-      p_cache.push_back(pos);
-    }
-
-    p_cacheVelocity.clear();
+//    p_hasVelocity = !p_cacheVelocity.empty();
 
     bool hasVelocityKey = isdPos.find("Velocities") != isdPos.end();
+    p_hasVelocity = hasVelocityKey; 
 
+    std::vector<ale::State> stateCache;
     if (hasVelocityKey) {
-      for (auto it = isdPos["Velocities"].begin(); it != isdPos["Velocities"].end(); it++) {
-        std::vector<double> vel = {it->at(0).get<double>(), it->at(1).get<double>(), it->at(2).get<double>()};
-        p_cacheVelocity.push_back(vel);
+      for (auto it = isdPos["Positions"].begin(); it != isdPos["Positions"].end(); it++) {
+        std::vector<double> pos = {it->at(0).get<double>(), it->at(1).get<double>(), it->at(2).get<double>()};
+        int index = it - isdPos["Positions"].begin();
+        std::vector<double> vel = isdPos["Velocities"][index];
```

Comment:
Why are the position and velocity being collected differently? You can use `.get<std::vector<double>>()` here too.

Reply:
Good question. I'll update this. Thanks for the feedback!

Reply:
If you're iteration over positions, specifically, does this work?

Reply:
Yeah, you can do `it->get<std::vector<double>>()`